### ISSUE TITLE
since there is REGEXP_TEST function, would it be better support regex replace?

### ISSUE THREAD

**Issue by dsonet** [arangodb/arangodb#1902]
Like introduce an REGEXP_REPLACE function or extend the SUBSTITUTE functionality to support this?

**Comment by jsteemann** [comment]
This is indeed planned for the future (and that's why the REGEXP function was renamed to REGEXP_TEST shortly before release).

**Comment by dsonet** [comment]
Good to know that. Thanks.

**Comment by jsteemann** [comment]
The `devel` branch has a (currently) non-documented and non-tested `REGEX_REPLACE` function.
The function signature is:
```
replaced = REGEX_REPLACE (value, regex, replacement, caseInsensitive)
```
Example:
```
RETURN REGEX_REPLACE('foo bar baz', '[a-z]{3}', 'bark')
```
While the function has already been implemented, documentation for it is missing in file `Documentation/Books/AQL/Functions/String.mdpp`. Additionally, test cases for it are missing in file `js/server/tests/aql/aql-functions-string.js`.
When documentation and test cases are ready, the function can be made official. Otherwise it will be removed in next release (3.2).

**Comment by Simran-B** [comment]
Are there any other flags, that could be allowed instead of just caseInsensitive as 4th parameter? Most notably the global flag is pretty common to not replace a single occurrence of a pattern only, but all (non-overlapping) occurrences of it.

I'll write the tests and documentation once this is decided.

**Comment by jsteemann** [comment]
I think it does global replacements by default and at the moment this can't be changed by setting any parameter.

**Comment by jsteemann** [comment]
Fixed by @akki for 3.2/devel.
